test/integration_test.go

```diff
@@ -604,7 +605,39 @@ func (s *IntegrationSuite) TestHints() {
 			assert.Equal(t, it.expectLen, len(data))
 		})
 	}
+}
+
+func (s *IntegrationSuite) TestMultipleHints() {
+	var (
+		db = s.DB()
+		t  = s.T()
+	)
 
+	type tt struct {
+		sql       string
+		args      []interface{}
+		expectLen int
+	}
+
+	for _, it := range []tt{
+		{"/*A! master */ /*A! fullscan */ SELECT * FROM student WHERE uid = 42 AND 1=2", nil, 0},
+		{"/*A! slave */ /*A! master */ /*A! fullscan */ SELECT id,name FROM student WHERE uid = ?", []interface{}{1}, 0},
+		{"/*A! master */ /*A! fullscan */ SELECT * FROM student WHERE uid = ?", []interface{}{1}, 1},
+		{"/*A! fullscan */ /*A! direct */ SELECT * FROM student WHERE uid in (?)", []interface{}{1}, 0},
```

Comment:
The full-scan UT should be like `select * from student where name like '%xxx%'`, because it will cause full-scan.
Meanwhile, these existing UT contains the shard key `uid`, it will never trigger full-scan.

Reply:
done